dev-docs/RULES.md

```
@@ -339,6 +339,189 @@ const notifyUser = (email: Email): void => emailService.sendWelcome(email);
+
+### The Fundamental Rule
+
+**You cannot test a component by mocking that component.**
+
+This seems obvious but is constantly violated. If you mock `EmojiFilter` to test `EmojiFilter`, you're not testing `EmojiFilter` at all.
+
+### Mock Decision Tree
+
+When deciding whether to mock something:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)